otterai/indexer.py

```diff
@@ -42,59 +45,78 @@ def index_codebase(root_dir: str) -> Dict[str, List[str]]:
         'test': [],
         'other': []
     }
+
+    default_ignore_patterns = [
+        '*.pyc', '__pycache__/*', '.git/*', '.github/*', 'node_modules/*',
+        '*.min.js', '*.min.css', '*.map', '*.lock', '*.sum',
+        'dist/*', 'build/*', '.env*', '*.log',
+        '*venv/*', '*.venv/*', '*.venv', 'venv/*', 'venv', '*.venv',
+        '*.pyc', '__pycache__/*', '.git/*', '.github/*', 'node_modules/*',
+    ]
     
     for root, _, files in os.walk(root_dir):
         for file in files:
             file_path = os.path.join(root, file)
             rel_path = os.path.relpath(file_path, root_dir)
             
             if should_ignore_file(rel_path):
+                print(f"Ignoring file: {file} because it matches ignore patterns.")
                 continue
                 
             file_type = get_file_type(rel_path) or 'other'
             index[file_type].append(rel_path)
     
     return index
 
-def analyze_project_structure(index: Dict[str, List[str]], repo_root: str) -> str:
+async def analyze_project_structure(index: Dict[str, List[str]], repo_root: str) -> str:
```

Comment:
🔄 The 'analyze_project_structure' function is now async. Ensure that all async functions are properly awaited to prevent blocking.